Fix Local Transcoding Demo

ericxtang opened the pull request:
2 things broke in the past few months -
* `BasicSubscriber` only accounts for a single callback function
* The `transcode` endpoint doesn't update the manifest after starting to transcode

This change addresses these 2 issues and changes the tests that break because of it.

TODO:
* Propagate changes for `vendor/github.com/livepeer/go-livepeer-basicnet/basic_network.go` and `vendor/github.com/livepeer/go-livepeer-basicnet/basic_subscriber.go` into `go-livepeer-basicnet`.

ericxtang committed:
make local transcoding demo work again

ericxtang committed:
fix tests

yondonfu commented on server/webserver.go:
```diff
@@ -41,12 +45,33 @@ func (s *LivepeerServer) StartWebserver() {
 		for i, vp := range ps {
 			vids[ids[i]] = vp
+			vParams := ffmpeg.VideoProfileToVariantParams(vp)
+			pl, _ := m3u8.NewMediaPlaylist(stream.DefaultHLSStreamWin, stream.DefaultHLSStreamCap)
```
Should we handle the error in case the media playlist creation fails?

yondonfu commented on server/webserver.go:
```diff
@@ -41,12 +45,33 @@ func (s *LivepeerServer) StartWebserver() {
+			variant := &m3u8.Variant{URI: fmt.Sprintf("%v.m3u8", ids[i]), Chunklist: pl, VariantParams: vParams}
+			manifest.Append(variant.URI, variant.Chunklist, variant.VariantParams)
 		}
+		s.LivepeerNode.VideoNetwork.UpdateMasterPlaylist(manifestID.String(), manifest)
```
This is specific to basicnet, but is there anything preventing any node from calling this function to update a given manifestID with *any* manifest?

ericxtang replied:
Good question.  Theoretically the node can check for the ID, but it's not doing that right now.  Filed an issue here.  https://github.com/livepeer/go-livepeer-basicnet/issues/25

yondonfu commented on vendor/github.com/livepeer/go-livepeer-basicnet/basic_subscriber.go:
```diff
@@ -102,33 +109,46 @@ func (s *BasicSubscriber) Subscribe(ctx context.Context, gotData func(seqNo uint
+					glog.V(common.DEBUG).Infof("Subscriber worker inserted segment: %v - took %v in total, %v waiting for data", msg.SeqNo, time.Since(start), networkWaitTime)
+				case <-ctxW.Done():
+					// s.networkStream = nil
+					s.working = false
```
Should we also set `s.running = false` at the end of this case since the for loop will exit?

yondonfu commented on vendor/github.com/livepeer/go-livepeer-basicnet/basic_subscriber.go:
```diff
@@ -20,10 +21,12 @@ var ErrSubscriber = errors.New("ErrSubscriber")
+	host         host.Host
+	msgChan      chan StreamDataMsg
+	callbacks    []func(uint64, []byte, bool)
+	running      bool
```
Do we still need the `working` bool if we have this `running` bool? It seems like they are both used to signal if the subscription worker for loop is running?

yondonfu reviewed (changes requested):
Left a few minor comments. Also, where do we currently set additional callback handlers for a single subscriber?

ericxtang committed:
remove , add error check

ericxtang commented:
Removed `working`, and added error check for when we are creating the MediaPlaylist in the webserver.

Here is the scenario where we have multiple callbacks:
* Start a local broadcast
* Start a local transcoding operation by calling `/transcode` - this creates a subscriber to the broadcasted stream
* Start watching the broadcasted stream.  This uses the same subscriber to the broadcasted stream.  If we don't support multiple callbacks, we have an indeterministic behavior (currently the channel alternates between the 2 go routines, but it could be anything...)

yondonfu commented:
Got it. 🚢
